packages/api/src/command/medical/document/start-bulk-get-doc-url.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  BulkGetDocumentsUrlProgress,
+  isBulkGetDocUrlProcessing,
+  BulkGetDocUrlStatus,
+} from "../../../domain/medical/bulk-get-document-url";
+import { Util } from "../../../shared/util";
+import { getPatientOrFail } from "../patient/get-patient";
+import { Patient } from "../../../domain/medical/patient";
+import { uuidv7 } from "@metriport/core/util/uuid-v7";
+import { storeBulkGetDocumentUrlQueryInit } from "../patient/bulk-get-doc-url-progress";
+import { makeLambdaClient } from "../../../external/aws/lambda";
+import { DocumentBulkSignerLambdaRequest } from "@metriport/core/external/aws/lambda-logic/document-bulk-signing";
+import { appendBulkGetDocUrlProgress } from "../patient/bulk-get-doc-url-progress";
+import { capture } from "../../../shared/notifications";
+
+const lambdaClient = makeLambdaClient();
+const bulkSigningLambdaName = "BulkUrlSigningLambda";
+
+/**
+ * The function `startBulkGetDocumentUrls` triggers the bulk signing process lambda for a patient's documents and
+ * returns the progress of the bulk signing.
+ * @param cxId - cxId
+ * @param patientId - patientId
+ * @returns a Promise that resolves to a BulkGetDocumentsUrlProgress object.
+ */
+export const startBulkGetDocumentUrls = async (
+  cxId: string,
+  patientId: string
+): Promise<BulkGetDocumentsUrlProgress> => {
+  const { log } = Util.out(`startBulkGetDocumentUrls - M patient ${patientId}`);
+  if (!bulkSigningLambdaName) throw new Error("Bulk Signing Lambda Name is undefined");
+  const patient = await getPatientOrFail({ id: patientId, cxId });
+
+  const bulkGetDocUrlProgress = patient.data.bulkGetDocumentsUrlProgress;
+
+  if (isBulkGetDocUrlProcessing(bulkGetDocUrlProgress?.status)) {
+    log(
+      `Patient ${patientId}, Request ${bulkGetDocUrlProgress?.requestId}, bulkGetDocUrlProgress is already 'processing', skipping...`
+    );
+    return createBulkGetDocumentUrlQueryResponse("processing", patient);
+  }
+
+  const requestId = getOrGenerateRequestId(bulkGetDocUrlProgress);
+
+  const updatedPatient = await storeBulkGetDocumentUrlQueryInit({
+    id: patient.id,
+    cxId: patient.cxId,
+    bulkGetDocumentsUrlProgress: { status: "processing" },
+    requestId,
+  });
+
+  const payload: DocumentBulkSignerLambdaRequest = {
+    patientId: patientId,
+    cxId: cxId,
+    requestId: requestId,
+  };
+
+  try {
+    lambdaClient
+      .invoke({
+        FunctionName: bulkSigningLambdaName,
+        InvocationType: "RequestResponse",
+        Payload: JSON.stringify(payload),
+      })
+      .promise();
+  } catch (error) {
+    appendBulkGetDocUrlProgress({
+      patient: { id: patientId, cxId: cxId },
+      status: "failed",
+      requestId: requestId,
+    });
+    capture.error(error, {
+      extra: { patientId, context: `startBulkGetDocumentUrls`, error },
+    });
+    return createBulkGetDocumentUrlQueryResponse("failed", updatedPatient);
+  }
```

Comment:
The error handling code has to be moved to `.catch()` on the promise, it doesn't work to use try/catch if we're not `awaiting` the Promise. Please test this.